dictation/router.js

```diff
@@ -0,0 +1,124 @@
+const { Router } = require("express");
+const {
+  Dictation,
+  Melody,
+  Sequelize: { Op }
+} = require("../models");
+const authMiddleware = require("../auth/middleware");
+const validaton = require("./validation");
+const router = new Router();
+
+// start a new dictation
+router.post(
+  "/melody/:melodyId/dictation",
+  authMiddleware,
+  async (req, res, next) => {
+    const { id: userId } = req.user;
+    const { melodyId } = req.params;
+    try {
+      const dictation = await Dictation.create({
+        melodyId,
+        userId
+      });
+      res.send(dictation);
+    } catch (error) {
+      next(error);
+    }
+  }
+);
+
+// get all dictations for this melody for this user
+router.get(
+  "/melody/:melodyId/dictation",
+  authMiddleware,
+  async (req, res, next) => {
+    const user = req.user;
+    try {
+      const dictation = await Dictation.findAll({
+        where: { melodyId: req.params.melodyId, userId: user.id }
+      });
+      res.send(dictation);
+    } catch (error) {
+      next(error);
+    }
+  }
+);
+// get stats for previous dictations for this melody for this user
+router.get(
+  "/melody/:melodyId/stats",
+  authMiddleware,
+  async (req, res, next) => {
+    const { id: userId } = req.user;
+    const { melodyId } = req.params;
+    try {
+      const all = await Dictation.count({
+        where: { melodyId, userId }
+      });
+      const finished = await Dictation.count({
+        where: {
+          melodyId,
+          userId,
+          score: { [Op.ne]: null }
+        }
+      });
+      // const successful = 100;
+      const successful = await Dictation.count({
+        where: {
+          melodyId,
+          userId,
+          score: 100
+        }
+      });
+      res.send({ all, finished, successful });
+    } catch (error) {
+      next(error);
+    }
+  }
+);
+
+// get a certain dictations (for this user)
+router.get(
+  "/melody/:melodyId/dictation/:dictationId",
+  authMiddleware,
+  async (req, res, next) => {
+    try {
+      const dictation = await Dictation.findByPk(req.params.dictationId);
+      res.send(dictation);
+    } catch (error) {
+      next(error);
+    }
+  }
+);
+
+// submit user input for certain dictation (for this user)
+router.put(
+  "/melody/:melodyId/dictation/:dictationId",
+  authMiddleware,
+  async (req, res, next) => {
+    const user = req.user;
+    try {
+      const dictation = await Dictation.findByPk(req.params.dictationId, {
+        include: Melody
+      });
+      if (dictation.userId === user.id) {
+        const result = validaton(dictation.Melody.abcNotes, req.body.userInput);
+        let scorePercent = Math.round(
```

Comment:
by @akellbl4:
Just use `const` for not changeable variables,
And if it just `score` that will be cleaner when you pass it in the hash.